### ISSUE TITLE
Does not work with Fastify v4

### ISSUE THREAD

**Issue by gajus** [DataDog/dd-trace-js#2117]
Simply starting server using `dd-trace/init` makes the server crash:

```
TypeError: next is not a function
    at /Users/gajus/Documents/dev/contra/contra-web-app/node_modules/dd-trace/packages/datadog-instrumentations/src/fastify.js:129:41
    at AsyncResource.runInAsyncScope (node:async_hooks:202:9)
    at Object.preParsing (/Users/gajus/Documents/dev/contra/contra-web-app/node_modules/dd-trace/packages/datadog-instrumentations/src/fastify.js:129:19)
    at next (/Users/gajus/Documents/dev/contra/contra-web-app/node_modules/fastify/lib/route.js:501:16)
    at preParsingHookRunner (/Users/gajus/Documents/dev/contra/contra-web-app/node_modules/fastify/lib/route.js:524:3)
    at runPreParsing (/Users/gajus/Documents/dev/contra/contra-web-app/node_modules/fastify/lib/route.js:475:5)
    at next (/Users/gajus/Documents/dev/contra/contra-web-app/node_modules/fastify/lib/hooks.js:168:7)
    at /Users/gajus/Documents/dev/contra/contra-web-app/node_modules/dd-trace/packages/datadog-instrumentations/src/fastify.js:62:29
    at AsyncResource.runInAsyncScope (node:async_hooks:202:9)
    at arguments.<computed> (/Users/gajus/Documents/dev/contra/contra-web-app/node_modules/dd-trace/packages/datadog-instrumentations/src/fastify.js:61:38)
    at Logger_1.Logger.adopt.nextMessage (/Users/gajus/Documents/dev/contra/contra-web-app/server/adoptDataDog.ts:8:7)
    at /Users/gajus/Documents/dev/contra/contra-web-app/node_modules/roarr/dist/src/factories/createLogger.js:198:20
    at AsyncLocalStorage.run (node:async_hooks:327:14)
    at Function.log.adopt (/Users/gajus/Documents/dev/contra/contra-web-app/node_modules/roarr/dist/src/factories/createLogger.js:192:34)
    at adoptDataDog (/Users/gajus/Documents/dev/contra/contra-web-app/server/adoptDataDog.ts:6:15)
    at Object.<anonymous> (/Users/gajus/Documents/dev/contra/contra-web-app/server/startHttpService.ts:85:17)
```

**Comment by NickDub-adeo** [comment]
Issue better explained in https://github.com/DataDog/dd-trace-js/issues/2118 😉

**Comment by rochdev** [comment]
This was fixed in #2114 and we're in the process of shipping a new patch release.
